Packs/Zscaler/Integrations/Zscaler/Zscaler.py

```diff
@@ -732,68 +772,108 @@ def sandbox_report(md5, details):
-    auth = login()
-    jsession_id = auth[:auth.index(';')]
-    DEFAULT_HEADERS['cookie'] = jsession_id
-
     LOG('command is %s' % (demisto.command(),))
-    try:
-        if demisto.command() == 'test-module':
-            # Checks if there is an authenticated session
-            http_request('GET', '/authenticatedSession', None, DEFAULT_HEADERS)
-            demisto.results('ok')
-        elif demisto.command() == 'url':
-            demisto.results(url_lookup(demisto.args()))
-        elif demisto.command() == 'ip':
-            demisto.results(ip_lookup(demisto.args()['ip']))
-        elif demisto.command() == 'zscaler-blacklist-url':
-            demisto.results(blacklist_url(demisto.args()['url']))
-        elif demisto.command() == 'zscaler-undo-blacklist-url':
-            demisto.results(unblacklist_url(demisto.args()['url']))
-        elif demisto.command() == 'zscaler-whitelist-url':
-            demisto.results(whitelist_url(demisto.args()['url']))
-        elif demisto.command() == 'zscaler-undo-whitelist-url':
-            demisto.results(unwhitelist_url(demisto.args()['url']))
-        elif demisto.command() == 'zscaler-blacklist-ip':
-            demisto.results(blacklist_ip(demisto.args()['ip']))
-        elif demisto.command() == 'zscaler-undo-blacklist-ip':
-            demisto.results(unblacklist_ip(demisto.args()['ip']))
-        elif demisto.command() == 'zscaler-whitelist-ip':
-            demisto.results(whitelist_ip(demisto.args()['ip']))
-        elif demisto.command() == 'zscaler-undo-whitelist-ip':
-            demisto.results(unwhitelist_ip(demisto.args()['ip']))
-        elif demisto.command() == 'zscaler-category-add-url':
-            demisto.results(category_add_url(demisto.args()['category-id'], demisto.args()['url']))
-        elif demisto.command() == 'zscaler-category-add-ip':
-            demisto.results(category_add_ip(demisto.args()['category-id'], demisto.args()['ip']))
-        elif demisto.command() == 'zscaler-category-remove-url':
-            demisto.results(category_remove_url(demisto.args()['category-id'], demisto.args()['url']))
-        elif demisto.command() == 'zscaler-category-remove-ip':
-            demisto.results(category_remove_ip(demisto.args()['category-id'], demisto.args()['ip']))
-        elif demisto.command() == 'zscaler-get-categories':
-            demisto.results(get_categories_command(demisto.args()['displayURL']))
-        elif demisto.command() == 'zscaler-get-blacklist':
-            demisto.results(get_blacklist_command())
-        elif demisto.command() == 'zscaler-get-whitelist':
-            demisto.results(get_whitelist_command())
-        elif demisto.command() == 'zscaler-sandbox-report':
-            demisto.results(sandbox_report_command())
-    except Exception as e:
-        LOG(str(e))
-        LOG.print_log()
-        raise
-    finally:
+
+    if demisto.command() == 'zscaler-login':
+        return_results(login_command())
+    else:
+        login()
```

Comment:
handle the case you are logging out, dont login and run test module if you are just trying to logout.